entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityHandler.kt

```diff
@@ -406,4 +418,16 @@ class EntityHandler(
             ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).contentType(MediaType.APPLICATION_JSON)
                 .body(InternalErrorResponse("An error occurred while deleting $attrId from $entityId"))
     }
+
+    private fun checkAttributesAreAuthorized(ngsiLdAttributes: List<NgsiLdAttribute>, entityUri: URI, userId: String) =
+        ngsiLdAttributes.forEach { ngsiLdAttribute ->
+            checkAttributeIsAuthorized(ngsiLdAttribute.name, entityUri, userId)
+        }
+
+    private fun checkAttributeIsAuthorized(expandedAttributeName: String, entityUri: URI, userId: String) {
+        if (expandedAttributeName == EGM_SPECIFIC_ACCESS_POLICY &&
+            !authorizationService.userIsAdminOfEntity(entityUri, userId)
+        )
+            throw AccessDeniedException("User forbidden to update access policy of entity")
```

Comment:
You can maybe add the entityId in the exception message, as done for other exceptions.

Reply:
indeed, done.